client/web/src/enterprise/executors/secrets/ExecutorSecretsListPage.tsx

```diff
@@ -0,0 +1,211 @@
+                    same name as a global secret will overwrite the global secret. Site admins can configure global
+                    secrets <Link to="/admin/executors/secrets">in site admin settings</Link>.
+                </>
+            }
+            connectionLoader={connectionLoader}
+            {...props}
+        />
+    )
+}
+
+export interface OrgExecutorSecretsListPageProps extends GlobalExecutorSecretsListPageProps {
+    orgID: Scalars['ID']
+}
+
+export const OrgExecutorSecretsListPage: React.FunctionComponent<
+    React.PropsWithChildren<OrgExecutorSecretsListPageProps>
+> = props => {
+    const connectionLoader = useCallback(
+        (scope: ExecutorSecretScope) => orgExecutorSecretsConnectionFactory(props.orgID, scope),
+        [props.orgID]
+    )
+    return (
+        <ExecutorSecretsListPage
+            namespaceID={props.orgID}
+            headerLine={
+                <>
+                    Configure executor secrets that will only be available to executions in this organization.
+                    <br />
+                    Global secrets are available to executions in this namespace. Secrets in this namespace with the
+                    same name as a global secret will overwrite the global secret. Site admins can configure global
+                    secrets <Link to="/admin/executors/secrets">in site admin settings</Link>.
+                </>
+            }
+            connectionLoader={connectionLoader}
+            {...props}
+        />
+    )
+}
+
+interface ExecutorSecretsListPageProps extends GlobalExecutorSecretsListPageProps {
+    namespaceID: Scalars['ID'] | null
+    headerLine: JSX.Element
+    connectionLoader: (scope: ExecutorSecretScope) => UseConnectionResult<ExecutorSecretFields>
+}
+
+const ExecutorSecretsListPage: React.FunctionComponent<React.PropsWithChildren<ExecutorSecretsListPageProps>> = ({
+    namespaceID,
+    headerLine,
+    connectionLoader,
+}) => {
+    const [selectedScope, setSelectedScope] = useState<ExecutorSecretScope>(ExecutorSecretScope.BATCHES)
+    const { loading, hasNextPage, fetchMore, connection, error, refetchAll } = connectionLoader(selectedScope)
+
+    const [showAddModal, setShowAddModal] = useState<boolean>(false)
+    const onClickAdd = useCallback<React.MouseEventHandler>(event => {
+        event.preventDefault()
+        setShowAddModal(true)
+    }, [])
+
+    const closeModal = useCallback(() => {
+        setShowAddModal(false)
+    }, [])
+    const afterAction = useCallback(() => {
+        setShowAddModal(false)
+        refetchAll()
+    }, [refetchAll])
+
+    return (
+        <>
+            <PageHeader
+                path={[{ text: 'Executor secrets' }]}
+                headingElement="h2"
+                description={headerLine}
+                actions={
+                    <>
+                        <Button onClick={onClickAdd} aria-label="Add new secret value" variant="primary">
```

Comment:
I think the button text is sufficient as a label here.

```suggestion
                        <Button onClick={onClickAdd} variant="primary">
```